onedal/linear_model/tests/test_ridge.py

```diff
@@ -0,0 +1,95 @@
+# ===============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+from daal4py.sklearn._utils import daal_check_version
+
+if daal_check_version((2024, "P", 600)):
+    import numpy as np
+    import pytest
+    from numpy.testing import assert_allclose, assert_array_equal
+    from sklearn.datasets import load_diabetes
+    from sklearn.metrics import mean_squared_error
+    from sklearn.model_selection import train_test_split
+
+    from onedal.linear_model import LinearRegression
+    from onedal.tests.utils._device_selection import get_queues
+
+    @pytest.mark.parametrize("queue", get_queues())
+    @pytest.mark.parametrize("dtype", [np.float32, np.float64])
+    def test_diabetes(queue, dtype):
+        X, y = load_diabetes(return_X_y=True)
+        X, y = X.astype(dtype), y.astype(dtype)
+        X_train, X_test, y_train, y_test = train_test_split(
+            X, y, train_size=0.8, random_state=777
+        )
+        model = LinearRegression(fit_intercept=True, alpha=0.1)
+        model.fit(X_train, y_train, queue=queue)
+        y_pred = model.predict(X_test, queue=queue)
+        assert mean_squared_error(y_test, y_pred) < 2396
```

Comment:
Is it possible to be more deterministic here? Even if dropping MSE is a good thing, might be good to throw an error.

Reply:
this is copy paste from [lin reg test](https://github.com/intel/scikit-learn-intelex/blob/2cd3be447f80da891c3275c9509d7b66ec2ab339/onedal/linear_model/tests/test_linear_regression.py#L30), not sure if modifying it makes sense. If that is not what you meant, could you please clarify exactly what part of this is not deterministic enough or where exactly an error should be thrown?

Reply:
If the MSE is 2395.567, could we test if it is 2395.567 with some tolerance instead of testing <2396? If the responses stays the same MSE also should stay the same apart from some numerical instability. Applies for lin reg as well.